Crash if data azurerm_private_dns_zone does not exist

<!---
Please note the following potential times when an issue might be in Terraform core:

* [Configuration Language](https://www.terraform.io/docs/configuration/index.html) or resource ordering issues
* [State](https://www.terraform.io/docs/state/index.html) and [State Backend](https://www.terraform.io/docs/backends/index.html) issues
* [Provisioner](https://www.terraform.io/docs/provisioners/index.html) issues
* [Registry](https://registry.terraform.io/) issues
* Spans resources across multiple providers

If you are running into one of these scenarios, we recommend opening an issue in the [Terraform core repository](https://github.com/hashicorp/terraform/) instead.
--->

<!--- Please keep this note for the community --->

### Community Note

* Please vote on this issue by adding a 👍 [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to the original issue to help the community and maintainers prioritize this request
* Please do not leave "+1" or "me too" comments, they generate extra noise for issue followers and do not help prioritize the request
* If you are interested in working on this issue or have submitted a pull request, please leave a comment

<!--- Thank you for keeping this note for the community --->

### Terraform (and AzureRM Provider) Version

```
Terraform v0.12.24
+ provider.azurerm v2.9.0
```

### Affected Resource(s)

* `data azurerm_private_dns_zone`

### Terraform Configuration Files

<!--- Information about code formatting: https://help.github.com/articles/basic-writing-and-formatting-syntax/#quoting-code --->

```hcl
data "azurerm_private_dns_zone" "foobar" {
  name = "foo"
  resource_group_name = "bar"
}
```

### Debug Output

<!---
Please provide a link to a GitHub Gist containing the complete debug output. Please do NOT paste the debug output in the issue; just paste a link to the Gist.

To obtain the debug output, see the [Terraform documentation on debugging](https://www.terraform.io/docs/internals/debugging.html).
--->

### Panic Output

https://gist.github.com/mjiderhamn/8767e81b6c51199188596bd4db439612

### Expected Behavior

If `data "azurerm_private_dns_zone"` is used for a non-existing Private DNS Zone, Terraform should inform that resource does not exist.

### Actual Behavior

If Private DNS Zone doesn't exist Terraform crashes.

### Important Factoids

In my case resource groups exists but Private DNS Zone does not.

Still happens with:
```
Terraform v0.12.28
+ provider.azurerm v2.16.0
```
```
panic: runtime error: invalid memory address or nil pointer dereference
2020-06-30T20:12:03.694+0200 [DEBUG] plugin.terraform-provider-azurerm_v2.16.0_x5: [signal SIGSEGV: segmentation violation code=0x1 addr=0x0 pc=0x45d3d52]
2020-06-30T20:12:03.694+0200 [DEBUG] plugin.terraform-provider-azurerm_v2.16.0_x5: 
2020-06-30T20:12:03.694+0200 [DEBUG] plugin.terraform-provider-azurerm_v2.16.0_x5: goroutine 64 [running]:
2020-06-30T20:12:03.694+0200 [DEBUG] plugin.terraform-provider-azurerm_v2.16.0_x5: github.com/terraform-providers/terraform-provider-azurerm/azurerm/internal/services/privatedns.dataSourceArmPrivateDnsZoneRead(0xc0009ea3f0, 0x4a1fd80, 0xc000404000, 0x0, 0x0)
2
```

--- a/azurerm/internal/services/privatedns/private_dns_zone_data_source.go
+++ b/azurerm/internal/services/privatedns/private_dns_zone_data_source.go
@@ -74,7 +74,7 @@ func dataSourceArmPrivateDnsZoneRead(d *schema.ResourceData, meta interface{}) e
 	if resourceGroup != "" {
 		zone, err := client.Get(ctx, resourceGroup, name)
 		if err != nil {
-			if utils.ResponseWasNotFound(resp.Response) {
+			if utils.ResponseWasNotFound(zone.Response) {
 				return fmt.Errorf("Private DNS Zone %q (Resource Group %q) was not found", name, resourceGroup)
 			}
 			return fmt.Errorf("reading Private DNS Zone %q (Resource Group %q): %+v", name, resourceGroup, err)

